src/pudl/metadata/classes.py

```diff
@@ -1632,3 +1633,45 @@ def to_sql(
                 check_values=check_values,
             )
         return metadata
+
+
+class CodeData(Base):
```

Comment:
Collecting all of the coding tables into a single data structure that we can refer to and iterate over makes sense to me. It's stored as a dictionary. How does it benefit from being a list here instead?

In the `Resource` and `Field` classes, we're using `from_XXX_ids()` rather than `from_XXX_names()` but they're doing the same thing, so I think we should use the same naming convention for the class methods.

There's already a Pydantic class defined here (`Encoder`) to contain the coding table information. Would it make sense to have this class contain a collection of `Encoder` objects instead? They'd get validated in the process too.

Would it be cleaner to add a `to_rst()` method to the `Encoder` class so that it can generate its own RST snippet? It would also need to know the directory to which the CSVs should be written so it can output the CSV file to generate the table from in the right place, and correctly reference that path in the RST it outputs.